modules/debug_cmd/debug_cmd.c

```diff
@@ -193,18 +200,23 @@ static void fileinfo_timeout(void *arg)
 	if (st->prm.ch && st->prm.srate)
 		s = ((double) st->sampc)  / st->prm.ch / st->prm.srate;
 
-	if (st->finished) {
-		info("debug_cmd: length = %1.3lf seconds\n", s);
+	if (re_atomic_rlx(&st->finished)) {
```

Comment:
Can you make the `atomic` related changes into a separate PR ?

Reply:
The bool `finished` may become obsolete.